geomesa-cassandra/geomesa-cassandra-datastore/src/main/scala/org/locationtech/geomesa/cassandra/data/CassandraDataStore.scala

```diff
@@ -0,0 +1,147 @@
+    val attributes = cols.map { c =>
+      val it = typeMap.inverse().get(c.getType)
+      attrTypeBuilder.binding(it).buildDescriptor(c.getName)
+    }
+    // TODO: allow user data to set dtg field
+    val dtgAttribute = attributes.find(_.getType.getBinding.isAssignableFrom(classOf[java.util.Date])).head
+    val sftBuilder = new SimpleFeatureTypeBuilder()
+    sftBuilder.addAll(attributes)
+    sftBuilder.setName(name.getLocalPart)
+    val sft = sftBuilder.buildFeatureType()
+    sft.getUserData.put(RichSimpleFeatureType.DEFAULT_DATE_KEY, dtgAttribute.getLocalName)
+    sft
+  }
+
+  sealed trait FieldSerializer {
+    def serialize(o: java.lang.Object): java.lang.Object
+    def deserialize(o: java.lang.Object): java.lang.Object
+  }
+  case object GeomSerializer extends FieldSerializer {
+    override def serialize(o: Object): AnyRef = {
+      val geom = o.asInstanceOf[Point]
+      ByteBuffer.wrap(WKBUtils.write(geom))
+    }
+
+    override def deserialize(o: Object): AnyRef = WKBUtils.read(o.asInstanceOf[ByteBuffer].array())
+  }
+
+  case object DefaultSerializer extends FieldSerializer {
+    override def serialize(o: Object): AnyRef = o
+    override def deserialize(o: Object): AnyRef = o
+  }
+
+  object FieldSerializer {
+    def apply(attrDescriptor: AttributeDescriptor): FieldSerializer = {
+      if(classOf[Geometry].isAssignableFrom(attrDescriptor.getType.getBinding)) GeomSerializer
+      else DefaultSerializer
+    }
+  }
+}
+
+class CassandraDataStore(session: Session, keyspaceMetadata: KeyspaceMetadata, ns: URI) extends ContentDataStore {
+  import scala.collection.JavaConversions._
+
+  override def createFeatureSource(contentEntry: ContentEntry): ContentFeatureSource =
+    new CassandraFeatureStore(contentEntry)
+
+  override def createSchema(featureType: SimpleFeatureType): Unit = {
+    val cols =
+      featureType.getAttributeDescriptors.map { ad =>
+        s"${ad.getLocalName}  ${CassandraDataStore.typeMap(ad.getType.getBinding).getName.toString}"
+      }.mkString(",")
+    val colCreate = s"(pkz int, z31 bigint, fid text, $cols, PRIMARY KEY (pkz, z31, fid))"
+    val stmt = s"create table ${featureType.getTypeName} $colCreate"
+    session.execute(stmt)
+  }
+
+
+  override def createContentState(entry: ContentEntry): ContentState =
+    new CassandraContentState(entry, session, keyspaceMetadata.getTable(entry.getTypeName))
+
+  override def createTypeNames(): util.List[Name] =
+    keyspaceMetadata.getTables.map { t => new NameImpl(ns.toString, t.getName) }.toList
+}
+
+object CassandraPrimaryKey {
+
+  case class Key(idx: Int, x: Double, y: Double, dk: Int, z: Int)
+
+  def unapply(idx: Int): Key = {
+    val dk = idx >> 16
+    val z = idx & 0x000000ff
+    val (x, y) = SFC2D.toPoint(z)
+    Key(idx, x, y, dk, z)
+  }
+
+  def apply(dtg: DateTime, x: Double, y: Double): Key = {
+    val dk = epochWeeks(dtg).getWeeks << 16
+    val z = SFC2D.toIndex(x, y).toInt
```

Comment:
could this .toInt cause any issues? Would it be safer to do a mask/shift or something?